### ISSUE TITLE
[Native] Serialization compiler throws exception after updating from 1.3.61 to 1.3.70

### ISSUE THREAD

**Issue by erikc5000** [Kotlin/kotlinx.serialization#744]
**Describe the bug**

After updating my project from 1.3.61 to 1.3.70 (and runtime to 0.20.0), I'm seeing an exception on native targets when compiling a module that contains custom serializers for classes defined in a different module of the project. This is the error from the build log:

```
2020-03-05T15:25:01.4198220Z > Task :extensions:serialization:compileKotlinIosArm64 FAILED
2020-03-05T15:25:01.4199780Z Resolved project :core in :extensions:serialization:iosArm64CompileKlibraries 
2020-03-05T15:25:01.4202450Z Resolved org.jetbrains.kotlin:kotlin-stdlib-common:1.3.70 in :extensions:serialization:iosArm64CompileKlibraries 
2020-03-05T15:25:01.4206520Z Resolved org.jetbrains.kotlinx:kotlinx-serialization-runtime-common:0.20.0 in :extensions:serialization:iosArm64CompileKlibraries 
2020-03-05T15:25:01.4208170Z Resolved org.jetbrains.kotlinx:kotlinx-serialization-runtime-native:0.20.0 in :extensions:serialization:iosArm64CompileKlibraries 
2020-03-05T15:25:01.4209740Z Resolved org.jetbrains.kotlin:kotlin-serialization-unshaded:1.3.70 in :extensions:serialization:kotlinNativeCompilerPluginClasspath 
2020-03-05T15:25:01.4210640Z w: ATTENTION!
2020-03-05T15:25:01.4211280Z This build uses unsafe internal compiler arguments:
2020-03-05T15:25:01.4211680Z 
2020-03-05T15:25:01.4212980Z -XXLanguage:+InlineClasses
2020-03-05T15:25:01.4213440Z 
2020-03-05T15:25:01.4214080Z This mode is not recommended for production use,
2020-03-05T15:25:01.4214800Z as no stability/compatibility guarantees are given on
2020-03-05T15:25:01.4215550Z compiler or generated code. Use it at your own risk!
2020-03-05T15:25:01.4215970Z 
2020-03-05T15:25:01.4225290Z e: Compilation failed: Symbol for deserialized  class Date is unbound
2020-03-05T15:25:01.4225790Z 
2020-03-05T15:25:01.4226810Z  * Source files: Date.kt, DateTime.kt, Instant.kt, OffsetDateTime.kt, OffsetTime.kt, Time.kt, TimeZone.kt, YearMonth.kt, ZonedDateTime.kt, Duration.kt, Period.kt, DateRange.kt, DateTimeInterval.kt, InstantInterval.kt, OffsetDateTimeInterval.kt, ZonedDateTimeInterval.kt
2020-03-05T15:25:01.4227880Z  * Compiler version info: Konan: 1.3.70 / Kotlin: 1.3.70
2020-03-05T15:25:01.4228810Z  * Output kind: LIBRARY
2020-03-05T15:25:01.4229240Z 
2020-03-05T15:25:01.4230470Z e: java.lang.IllegalStateException: Symbol for deserialized  class Date is unbound
2020-03-05T15:25:01.4231460Z 	at org.jetbrains.kotlin.ir.symbols.impl.IrBindableSymbolBase.getOwner(IrSymbolBase.kt:44)
2020-03-05T15:25:01.4232460Z 	at org.jetbrains.kotlinx.serialization.compiler.backend.ir.SerializerIrGenerator.<init>(SerializerIrGenerator.kt:43)
2020-03-05T15:25:01.4233510Z 	at org.jetbrains.kotlinx.serialization.compiler.backend.ir.SerializerIrGenerator$Companion.generate(SerializerIrGenerator.kt:469)
2020-03-05T15:25:01.4234570Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializerClassLowering.lower(SerializationLoweringExtension.kt:48)
2020-03-05T15:25:01.4235680Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitClass(SerializationLoweringExtension.kt:34)
2020-03-05T15:25:01.4236830Z 	at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitClass(IrElementVisitorVoid.kt:44)
2020-03-05T15:25:01.4237930Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitClass(SerializationLoweringExtension.kt:28)
2020-03-05T15:25:01.4239060Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitClass(SerializationLoweringExtension.kt:28)
2020-03-05T15:25:01.4240580Z 	at org.jetbrains.kotlin.ir.declarations.impl.IrClassImpl.accept(IrClassImpl.kt:91)
2020-03-05T15:25:01.4241540Z 	at org.jetbrains.kotlin.ir.declarations.impl.IrFileImpl.acceptChildren(IrFileImpl.kt:66)
2020-03-05T15:25:01.4243760Z 	at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoidKt.acceptChildrenVoid(IrElementVisitorVoid.kt:271)
2020-03-05T15:25:01.4245180Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitElement(SerializationLoweringExtension.kt:30)
2020-03-05T15:25:01.4246430Z 	at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitPackageFragment(IrElementVisitorVoid.kt:30)
2020-03-05T15:25:01.4247780Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitPackageFragment(SerializationLoweringExtension.kt:28)
2020-03-05T15:25:01.4248890Z 	at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitFile(IrElementVisitorVoid.kt:37)
2020-03-05T15:25:01.4250380Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitFile(SerializationLoweringExtension.kt:28)
2020-03-05T15:25:01.4251470Z 	at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitFile(IrElementVisitorVoid.kt:38)
2020-03-05T15:25:01.4252550Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitFile(SerializationLoweringExtension.kt:28)
2020-03-05T15:25:01.4253670Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitFile(SerializationLoweringExtension.kt:28)
2020-03-05T15:25:01.4254640Z 	at org.jetbrains.kotlin.ir.declarations.impl.IrFileImpl.accept(IrFileImpl.kt:63)
2020-03-05T15:25:01.4255580Z 	at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoidKt.acceptVoid(IrElementVisitorVoid.kt:267)
2020-03-05T15:25:01.4256620Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt.runOnFileInOrder(SerializationLoweringExtension.kt:28)
2020-03-05T15:25:01.4257860Z 	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtension.generate(SerializationLoweringExtension.kt:60)
2020-03-05T15:25:01.4259160Z 	at org.jetbrains.kotlin.backend.konan.ToplevelPhasesKt$psiToIrPhase$1$1.invoke(ToplevelPhases.kt:159)
2020-03-05T15:25:01.4260150Z 	at org.jetbrains.kotlin.backend.konan.ToplevelPhasesKt$psiToIrPhase$1$1.invoke(ToplevelPhases.kt)
2020-03-05T15:25:01.4261080Z 	at org.jetbrains.kotlin.psi2ir.Psi2IrTranslator.postprocess(Psi2IrTranslator.kt:93)
2020-03-05T15:25:01.4262220Z 	at org.jetbrains.kotlin.psi2ir.Psi2IrTranslator.generateModuleFragment(Psi2IrTranslator.kt:78)
2020-03-05T15:25:01.4263230Z 	at org.jetbrains.kotlin.backend.konan.ToplevelPhasesKt$psiToIrPhase$1.invoke(ToplevelPhases.kt:215)
2020-03-05T15:25:01.4264210Z 	at org.jetbrains.kotlin.backend.konan.ToplevelPhasesKt$psiToIrPhase$1.invoke(ToplevelPhases.kt)
2020-03-05T15:25:01.4265210Z 	at org.jetbrains.kotlin.backend.common.phaser.PhaseBuildersKt$namedOpUnitPhase$1.invoke(PhaseBuilders.kt:149)
2020-03-05T15:25:01.4266830Z 	at org.jetbrains.kotlin.backend.common.phaser.PhaseBuildersKt$namedOpUnitPhase$1.invoke(PhaseBuilders.kt:147)
2020-03-05T15:25:01.4267890Z 	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper$runBody$1.invoke(CompilerPhase.kt:128)
2020-03-05T15:25:01.4268880Z 	at org.jetbrains.kotlin.backend.common.phaser.CompilerPhaseKt.downlevel(CompilerPhase.kt:24)
2020-03-05T15:25:01.4269860Z 	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper.runBody(CompilerPhase.kt:127)
2020-03-05T15:25:01.4270850Z 	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper.invoke(CompilerPhase.kt:105)
2020-03-05T15:25:01.4271810Z 	at org.jetbrains.kotlin.backend.common.phaser.CompositePhase.invoke(PhaseBuilders.kt:30)
2020-03-05T15:25:01.4272790Z 	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper$runBody$1.invoke(CompilerPhase.kt:128)
2020-03-05T15:25:01.4274020Z 	at org.jetbrains.kotlin.backend.common.phaser.CompilerPhaseKt.downlevel(CompilerPhase.kt:24)
2020-03-05T15:25:01.4275000Z 	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper.runBody(CompilerPhase.kt:127)
2020-03-05T15:25:01.4276260Z 	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper.invoke(CompilerPhase.kt:105)
2020-03-05T15:25:01.4277230Z 	at org.jetbrains.kotlin.backend.common.phaser.CompilerPhaseKt.invokeToplevel(CompilerPhase.kt:42)
2020-03-05T15:25:01.4278170Z 	at org.jetbrains.kotlin.backend.konan.KonanDriverKt.runTopLevelPhases(KonanDriver.kt:28)
2020-03-05T15:25:01.4279030Z 	at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:77)
2020-03-05T15:25:01.4279940Z 	at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:34)
2020-03-05T15:25:01.4280890Z 	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:85)
2020-03-05T15:25:01.4281740Z 	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:43)
2020-03-05T15:25:01.4282550Z 	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:104)
2020-03-05T15:25:01.4283330Z 	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:82)
2020-03-05T15:25:01.4284110Z 	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:50)
2020-03-05T15:25:01.4284950Z 	at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMainNoExit(CLITool.kt:215)
2020-03-05T15:25:01.4285860Z 	at org.jetbrains.kotlin.cli.bc.K2Native$Companion$mainNoExit$1.invoke(K2Native.kt:246)
2020-03-05T15:25:01.4286790Z 	at org.jetbrains.kotlin.cli.bc.K2Native$Companion$mainNoExit$1.invoke(K2Native.kt:238)
2020-03-05T15:25:01.4287620Z 	at org.jetbrains.kotlin.util.UtilKt.profileIf(Util.kt:27)
2020-03-05T15:25:01.4288380Z 	at org.jetbrains.kotlin.util.UtilKt.profile(Util.kt:21)
2020-03-05T15:25:01.4289190Z 	at org.jetbrains.kotlin.cli.bc.K2Native$Companion.mainNoExit(K2Native.kt:245)
2020-03-05T15:25:01.4290040Z 	at org.jetbrains.kotlin.cli.bc.K2NativeKt.mainNoExit(K2Native.kt:376)
2020-03-05T15:25:01.4290880Z 	at org.jetbrains.kotlin.cli.utilities.MainKt$daemonMain$1.invoke(main.kt:39)
2020-03-05T15:25:01.4291740Z 	at org.jetbrains.kotlin.cli.utilities.MainKt$daemonMain$1.invoke(main.kt)
2020-03-05T15:25:01.4292570Z 	at org.jetbrains.kotlin.cli.utilities.MainKt.mainImpl(main.kt:17)
2020-03-05T15:25:01.4293400Z 	at org.jetbrains.kotlin.cli.utilities.MainKt.daemonMain(main.kt:39)
2020-03-05T15:25:01.4294200Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-03-05T15:25:01.4295120Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-03-05T15:25:01.4296280Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-03-05T15:25:01.4297260Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-03-05T15:25:01.4298130Z 	at org.jetbrains.kotlin.compilerRunner.KonanCliRunner.run(KotlinNativeToolRunner.kt:178)
2020-03-05T15:25:01.4299100Z 	at org.jetbrains.kotlin.gradle.tasks.AbstractKotlinNativeCompile.compile(KotlinNativeTasks.kt:278)
2020-03-05T15:25:01.4299940Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-03-05T15:25:01.4300780Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-03-05T15:25:01.4301700Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-03-05T15:25:01.4302530Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-03-05T15:25:01.4303310Z 	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
2020-03-05T15:25:01.4304290Z 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)
2020-03-05T15:25:01.4305300Z 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
2020-03-05T15:25:01.4306410Z 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
2020-03-05T15:25:01.4307370Z 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:727)
2020-03-05T15:25:01.4308310Z 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:694)
2020-03-05T15:25:01.4309300Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:568)
2020-03-05T15:25:01.4310600Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
2020-03-05T15:25:01.4311710Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
2020-03-05T15:25:01.4312780Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
2020-03-05T15:25:01.4313820Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
2020-03-05T15:25:01.4314830Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
2020-03-05T15:25:01.4315820Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
2020-03-05T15:25:01.4316830Z 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
2020-03-05T15:25:01.4317880Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:553)
2020-03-05T15:25:01.4318990Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:536)
2020-03-05T15:25:01.4320040Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:109)
2020-03-05T15:25:01.4321130Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:276)
2020-03-05T15:25:01.4322230Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:265)
2020-03-05T15:25:01.4323210Z 	at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:33)
2020-03-05T15:25:01.4324030Z 	at java.util.Optional.orElseGet(Optional.java:267)
2020-03-05T15:25:01.4324830Z 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:33)
2020-03-05T15:25:01.4325710Z 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
2020-03-05T15:25:01.4334200Z 	at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
2020-03-05T15:25:01.4335400Z 	at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
2020-03-05T15:25:01.4336450Z 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
2020-03-05T15:25:01.4337450Z 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
2020-03-05T15:25:01.4338420Z 	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
2020-03-05T15:25:01.4339380Z 	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
2020-03-05T15:25:01.4340270Z 	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
2020-03-05T15:25:01.4341200Z 	at org.gradle.internal.execution.steps.CatchExceptionStep.execute(CatchExceptionStep.java:34)
2020-03-05T15:25:01.4342150Z 	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
2020-03-05T15:25:01.4343110Z 	at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
2020-03-05T15:25:01.4344050Z 	at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
2020-03-05T15:25:01.4345040Z 	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:49)
2020-03-05T15:25:01.4346020Z 	at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:153)
2020-03-05T15:25:01.4346890Z 	at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:67)
2020-03-05T15:25:01.4347750Z 	at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:41)
2020-03-05T15:25:01.4349020Z 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
2020-03-05T15:25:01.4350030Z 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
2020-03-05T15:25:01.4350990Z 	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
2020-03-05T15:25:01.4351920Z 	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
2020-03-05T15:25:01.4352870Z 	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
2020-03-05T15:25:01.4353840Z 	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
2020-03-05T15:25:01.4354670Z 	at java.util.Optional.map(Optional.java:215)
2020-03-05T15:25:01.4355490Z 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
2020-03-05T15:25:01.4356430Z 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
2020-03-05T15:25:01.4357350Z 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
2020-03-05T15:25:01.4358280Z 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
2020-03-05T15:25:01.4359380Z 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
2020-03-05T15:25:01.4360460Z 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
2020-03-05T15:25:01.4361470Z 	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
2020-03-05T15:25:01.4362460Z 	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
2020-03-05T15:25:01.4363480Z 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
2020-03-05T15:25:01.4364530Z 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
2020-03-05T15:25:01.4365460Z 	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
2020-03-05T15:25:01.4366560Z 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
2020-03-05T15:25:01.4367440Z 	at java.util.Optional.orElseGet(Optional.java:267)
2020-03-05T15:25:01.4368270Z 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
2020-03-05T15:25:01.4369220Z 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
2020-03-05T15:25:01.4370240Z 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
2020-03-05T15:25:01.4371260Z 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
2020-03-05T15:25:01.4372240Z 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
2020-03-05T15:25:01.4373200Z 	at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
2020-03-05T15:25:01.4374220Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:192)
2020-03-05T15:25:01.4375250Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:184)
2020-03-05T15:25:01.4376260Z 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
2020-03-05T15:25:01.4377310Z 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
2020-03-05T15:25:01.4378370Z 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
2020-03-05T15:25:01.4379650Z 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
2020-03-05T15:25:01.4380660Z 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
2020-03-05T15:25:01.4381680Z 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
2020-03-05T15:25:01.4382700Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
2020-03-05T15:25:01.4383690Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
2020-03-05T15:25:01.4384680Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
2020-03-05T15:25:01.4385720Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
2020-03-05T15:25:01.4386820Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
2020-03-05T15:25:01.4387850Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
2020-03-05T15:25:01.4388960Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
2020-03-05T15:25:01.4389990Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
2020-03-05T15:25:01.4391000Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
2020-03-05T15:25:01.4392010Z 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
2020-03-05T15:25:01.4393030Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
2020-03-05T15:25:01.4394000Z 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
2020-03-05T15:25:01.4395020Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
2020-03-05T15:25:01.4396240Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
2020-03-05T15:25:01.4397370Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
2020-03-05T15:25:01.4398470Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
2020-03-05T15:25:01.4399500Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
2020-03-05T15:25:01.4400510Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
2020-03-05T15:25:01.4401540Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
2020-03-05T15:25:01.4402520Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
2020-03-05T15:25:01.4403500Z 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2020-03-05T15:25:01.4404450Z 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
2020-03-05T15:25:01.4405340Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2020-03-05T15:25:01.4406340Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2020-03-05T15:25:01.4407290Z 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
2020-03-05T15:25:01.4408110Z 	at java.lang.Thread.run(Thread.java:748)
```
This is only happening on native -- JVM appears to be fine.

**To Reproduce**

This pull request shows the changes and build error: https://github.com/erikc5000/island-time/pull/42

Full source code: https://github.com/erikc5000/island-time/tree/kotlin-1.3.70

To reproduce, compile the `:extensions:serialization` project on any native target.

**Expected behavior**

Successful compilation? It did work in 1.3.61.

**Environment**
 - Kotlin version: [1.3.70]
 - Library version: [0.20.0]
 - Kotlin platforms: [Native]
 - Gradle version: [6.2.1]

**Comment by sandwwraith** [comment]
As a workaround, you may try to erase all `@Serializer(forClass = Time::class)` constructions: they're only required for autogenerating `serialize` and `deserialize`, and you've wrote them manually.

**Comment by erikc5000** [comment]
That does the trick. Thanks!

There's no mention in the docs on custom serializers that the annotation can be omitted and the java.util.Date example in there uses it, so I assumed it was required. It might be a good idea to mention that it's only needed when you want to autogenerate one or more methods.

**Comment by sandwwraith** [comment]
Will be fixed in 1.3.71
